### PR TITLE
ci(circle-ci): install gke auth plugin

### PR DESCRIPTION
**Pull request by loksonarius**
Kubectl update has deprecated the previous gcloud auth plugin. The new plugin requires direct installation on the system for kubectl to auth against gke clusters. This commit adds a step to install the new plugin during gke cluster provisioning.

## Test Plan

Run CI for the PR and see if gke e2e tests work fine.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -710,6 +710,9 @@ commands:
           gcloud-service-key: <<parameters.gcloud-service-key>>
           google-compute-zone: <<parameters.google-compute-zone>>
           google-project-id: <<parameters.google-project-id>>
+      - run:
+          command: gcloud components install gke-gcloud-auth-plugin
```
Comment (stoksc) [inline .circleci/config.yml:714]: Kicked off CI in an effort to maybe save you time, now it says
```
ERROR: (gcloud.components.install) The following components are not available for your current SDK version [gke-gcloud-auth-plugin]. Please run `gcloud components update` to update your SDK.
```

Comment (dzhu) [reply .circleci/config.yml:714]: The Google Cloud SDK also needs to be upgraded. Though I tried that and then it still doesn't find the plugin for some reason.

**Comment by loksonarius** [comment]
Looking good? @ioga @dzhu @stoksc 
https://app.circleci.com/pipelines/github/determined-ai/determined/30890/workflows/2251f450-a16e-4762-b1c6-2835d49f70f2/jobs/1064402

**Review by ioga — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -710,6 +710,12 @@ commands:
           gcloud-service-key: <<parameters.gcloud-service-key>>
           google-compute-zone: <<parameters.google-compute-zone>>
           google-project-id: <<parameters.google-project-id>>
+      - run:
+          command: |
+            gcloud components update --quiet
```
Comment (dzhu) [inline .circleci/config.yml:715]: You should probably change the version on installation rather than this separate upgrade step.

Comment (loksonarius) [reply .circleci/config.yml:715]: hehe, yeah that prolly makes more sense.
I'll make the change, let CI run up the cluster deployment and merge once that passes if that's cool

**Pull request merged**